src/lib/types/Payload.ts

```diff
@@ -151,4 +151,5 @@ export namespace Payload {
   }
 
   export type Hook<Value, R = boolean> = (value: Value) => Awaitable<R>;
+  export type HookWithKey<Value, R = boolean> = (value: Value, key: string) => Awaitable<R>;
```

Comment:
```suggestion

  export type HookWithKey<Value, R = boolean> = (value: Value, key: string) => Awaitable<R>;
```

The `R` type generic should be `unknown` when being used so make sure to add the second type generic parameter everywhere.

Reply:
I don't fully understand what R is...